### PR TITLE
Use DmaStream for virtio-block driver

### PR DESCRIPTION
**Pull request by liqinggd**

### CODE REVIEW

File: framework/aster-frame/src/vm/dma/dma_stream.rs [hunk 1]
```diff
@@ -222,6 +222,73 @@ impl HasPaddr for DmaStream {
     }
 }
 
+/// A slice of streaming DMA mapping.
+#[derive(Debug, Clone)]
+pub struct DmaStreamSlice {
+    stream: DmaStream,
```
Comment (tatetian) [inline framework/aster-frame/src/vm/dma/dma_stream.rs:228]: `DmaStreamSlice` is so similar to `DmaStream` that adding the concept of `DmaStreamSlice` only provides limited marginal value. Furthermore, the standard "slice" in Rust means borrowed data, not owned data. This is in conflict with the semantic of `DmaStreamSlice`, which is stilled owned.

For the reasons above, I propose re-defining `DmaStreamSlice` as below.

```rust
pub struct DmaStreamSlice<'a> {
    stream: &'a DmaStream,
    offset: usize,
    len: usize,
}
```

**Comment by liqinggd** [comment]
@StevenJiang1110 could modify the first commit about `DmBuf` trait?

File: framework/aster-frame/src/vm/dma/dma_stream.rs (hunk 1 above)
Comment (liqinggd) [reply framework/aster-frame/src/vm/dma/dma_stream.rs]: Done

File: kernel/comps/virtio/src/dma_buf.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    fn addr(&self) -> Daddr {
+        self.daddr()
+    }
+
+    fn len(&self) -> usize {
+        self.nframes() * PAGE_SIZE
+    }
+}
+
+impl DmaBuf for (Daddr, usize) {
```
[the remark this answers is not in the document]
Comment (liqinggd) [reply kernel/comps/virtio/src/dma_buf.rs]: Resoved in https://github.com/asterinas/asterinas/pull/647

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**